### PR TITLE
{lib}[foss/2020b] TensorFlow v2.2.3

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    ('Markdown', '3.3.4', {
+        'checksums': ['31b5b491868dcc87d6c24b7e3d19a0d730d59d3e46f4eea6430a321bed387a49'],
+    }),
+    ('pyasn1-modules', '0.2.8', {
+        'checksums': ['905f84c712230b2c592c19470d3ca8d552de726050d1d1716282a1f6146be65e'],
+    }),
+    ('rsa', '4.7.2', {
+        'checksums': ['9d689e6ca1b3038bc82bf8d23e944b6b6037bc02301a574935b2dd946e0353b9'],
+    }),
+    ('cachetools', '4.2.1', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:64]:
```suggestion
    ('cachetools', '4.2.2', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        'checksums': ['f469e29e7aa4cff64d8de4aad95ce76de8ea1125a16c68e0d93f65c3c3dc92e9'],
+    }),
+    ('google-auth', '1.28.0', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:67]:
```suggestion
    ('google-auth', '1.35.0', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        'modulename': 'google.auth',
+        'checksums': ['9bd436d19ab047001a1340720d2b629eb96dd503258c524921ec2af3ee88a80e'],
+    }),
+    ('oauthlib', '3.1.0', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:71]:
```suggestion
    ('oauthlib', '3.1.1', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        'checksums': ['bee41cc35fcca6e988463cacc3bcb8a96224f470ca547e697b604cc697b2f889'],
+    }),
+    ('requests-oauthlib', '1.3.0', {
+        'checksums': ['b4261601a71fd721a8bd6d7aa1cc1d6a8a93b4a9f5e96626f8e4d91e8beeaa6a'],
+    }),
+    ('google-auth-oauthlib', '0.4.4', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:77]:
```suggestion
    ('google-auth-oauthlib', '0.4.5', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        'checksums': ['09832c6e75032f93818edf1affe4746121d640c625a5bef9b5c96af676e98eee'],
+    }),
+    ('Werkzeug', '1.0.1', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:80]:
```suggestion
    ('Werkzeug', '2.0.1', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        'checksums': ['6c80b1e5ad3665290ea39320b91e1be1e0d5f60652b964a3070216de83d2e47c'],
+    }),
+    ('absl-py', '0.12.0', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:83]:
```suggestion
    ('absl-py', '0.13.0', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+        'modulename': 'absl',
+        'checksums': ['b44f68984a5ceb2607d135a615999b93924c771238a63920d17d3387b0d229d5'],
+    }),
+    ('astunparse', '1.6.3', {
+        'checksums': ['5ad93a8456f0d084c3456d059fd9a92cce667963232cbf763eac3bc5b7940872'],
+    }),
+    ('grpcio', '1.36.1', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:90]:
```suggestion
    ('grpcio', '1.39.0', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        'modulename': 'grpc',
+        'checksums': ['a66ea59b20f3669df0f0c6a3bd57b985e5b2d1dcf3e4c29819bb8dc232d0fd38'],
+    }),
+    ('gviz-api', '1.9.0', {
+        'source_tmpl': 'gviz_api-%(version)s.tar.gz',
+        'checksums': ['43d13ccc21834d0501b33a291ef3265e933dbb4bbdca3d34b1ed0a048c0ef640'],
+    }),
+    ('tensorboard', version, {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:98]:
```suggestion
    ('tensorboard', '2.2.2', {
```

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+    }),
+    ('termcolor', '1.1.0', {
+        'checksums': ['1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b'],
+    }),
+    ('tensorflow-estimator', '2.2.0', {
+        'source_tmpl': 'tensorflow_estimator-%(version)s-py2.py3-none-any.whl',
+        'unpack_sources': False,
+        'checksums': ['d09dacdd127f2579cea8d5af21f4a918036b8ae246adc82f26b61f91cc247dc2'],
+    }),
+    ('astor', '0.8.1', {
```
Comment (Flamefire) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:124]: Not required at all as far as I can tell. Not sure why this is included...

**Review by Flamefire — CHANGES_REQUESTED** [review]
See comments inline. Need updating checksums after changing versions too.

Note that you can use "Add to batch commit" (or so) from the "Files" tab to accept all changes in 1 commit

**Comment by Flamefire** [comment]
Required Bazel at https://github.com/easybuilders/easybuild-easyconfigs/pull/13855

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb (hunk 9 above)
Comment (Flamefire) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:124]: Ah no, required by TF

**Comment by sassy-crick** [comment]
Dependencies:

Requires [Bazel](https://github.com/easybuilders/easybuild-easyconfigs/pull/13855)

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+    ('nsync', '1.24.0'),
+    ('SQLite', '3.33.0'),
+    ('PCRE', '8.44'),
+    ('protobuf-python', '3.14.0'),
+    ('flatbuffers-python', '1.12'),
+    ('typing-extensions', '3.7.4.3'),
+    ('libpng', '1.6.37'),
+    ('snappy', '1.1.8'),
+    ('zlib', '1.2.11'),
+    # TF 2.2 requires SciPy 1.4.1 due to potential crashes with other versions
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.2.2-foss-2020b.eb:44]: Huh, indeed, see also https://github.com/tensorflow/tensorflow/pull/41866 .

**Comment by Flamefire** [comment]
Tests started, @sassy-crick can you mark this as ready for review?

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusi8012 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/c6bd29458696ded2f74a480e14aaedd5 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusml14 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 2.7.5
See https://gist.github.com/f695ad69f41a1611d63e45b83cb86615 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/f29af09d41697a0f21657b8021e4aadd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3504.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/1d3c4083fe5d6728003419e183da04b9 for a full test report.

**Comment by boegel** [comment]
@Flamefire We probably shouldn't let the failing test report from POWER9 block this PR?

```
Executed 15 out of 746 tests: 15 tests pass, 1 fails to build and 730 were skipped.
FAILED: Build did NOT complete successfully
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2642.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/b38fc456478ad34900b0da01ae6480bb for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**